source/common/common/logger.h

```diff
@@ -417,33 +419,40 @@ template <Id id> class Loggable {
 
 #define ENVOY_LOG_EVERY_NTH(LEVEL, N, ...)                                                         \
   do {                                                                                             \
-    static auto* count = new std::atomic<uint64_t>();                                              \
-    if ((count->fetch_add(1) % N) == 0) {                                                          \
-      ENVOY_LOG(LEVEL, ##__VA_ARGS__);                                                             \
+    if (ENVOY_LOG_COMP_LEVEL(ENVOY_LOGGER(), LEVEL)) {                                             \
+      static auto* count = new std::atomic<uint64_t>();                                            \
+      if ((count->fetch_add(1) % N) == 0) {                                                        \
+        ENVOY_LOG(LEVEL, ##__VA_ARGS__);                                                           \
+      }                                                                                            \
     }                                                                                              \
   } while (0)
 
 #define ENVOY_LOG_EVERY_POW_2(LEVEL, ...)                                                          \
   do {                                                                                             \
-    static auto* count = new std::atomic<uint64_t>();                                              \
-    if (std::bitset<64>(1 + count->fetch_add(1)).count() == 1) {                                   \
-      ENVOY_LOG(LEVEL, ##__VA_ARGS__);                                                             \
+    if (ENVOY_LOG_COMP_LEVEL(ENVOY_LOGGER(), LEVEL)) {                                             \
+      static auto* count = new std::atomic<uint64_t>();                                            \
+      if (std::bitset<64>(1 + count->fetch_add(1)).count() == 1) {                                 \
```

Comment:
How about this kind of thing:
```
auto bits = count->fetch_add(1);
if ( bits && !(bits & (bits-1)) )
```
The `unsigned` type here is fine.
Any bitwise `&&` of any power of 2 with a number less by 1 will give 0. No need to involve `std::bitset` and we get rid of that magic number too.

Reply:
I went with `std::bitset because` I think I remember it being compiled to something fast under the hood (__builtin_popcount?); but also - and this is probably subjective - I find it easier to grok. The +1 deserves a comment, e.g. `// +1 because the first hit needs to be logged`, but this imho this also wouldn't hurt with the `-1` (or maybe the whole expression) in the variant you propose, depending on who the audience is?

Reply:
Ok. Let's leave it be for now, no problemo. `bitset.count()` is readable, true.
Yes, `__builtin_popcount` for GCC and for LLVM it's a similar thing; which translates, depending on the platform, into about 2-3 cpu intructions, or into Kernighan’s algorithm (latter is slower than this little trick).

But you're right. To follow readibility guidelines i should've suggested a lambda at least here, sth like `auto is_exactly_one_bit_set = [](const uint64_t bits) { return bits && !(bits & (bits-1)); };`.